File: core/src/main/java/com/larryhsiao/nyx/core/jots/filter/WrappedFilter.java
package com.larryhsiao.nyx.core.jots.filter;

/**
 * Wrapped object of {@link Filter}
 */
public class WrappedFilter implements Filter {
    private final Filter filter;

    public WrappedFilter(Filter filter) {
        this.filter = filter;
    }

    @Override
    public String keyword() {
        return filter.keyword();
    }

    @Override
    public long[] dateRange() {
        return filter.dateRange();
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/filter/WrappedFilter.kt
package com.larryhsiao.nyx.core.jots.filter

/**
 * Wrapped object of [Filter]
 */
open class WrappedFilter(private val filter: Filter) : Filter {
    override fun keyword(): String? {
        return filter.keyword()
    }

    override fun dateRange(): LongArray? {
        return filter.dateRange()
    }

    override fun ids(): LongArray? {
        return filter.ids()
    }
}